app/src/main/java/org/mozilla/fenix/historymetadata/HistoryMetadataService.kt

```diff
@@ -127,6 +136,18 @@ class DefaultHistoryMetadataService(
             storage.deleteHistoryMetadataOlderThan(olderThan)
         }
     }
+
+    override fun removeMetadata(groups: Set<HistoryMetadataGroup>) {
+        scope.launch(IO) {
+            groups.forEach { group ->
+                try {
+                    storage.deleteHistoryMetadata(group.title)
+                } catch (e: Exception) {
```

Comment:
No need to wrap this in try/catch - internally these calls are already wrapped in selective try/catch, see https://github.com/mozilla-mobile/android-components/blob/da9e3883f9a5c8720e68dfcd14a75dc00349f5bb/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt#L311 and https://github.com/mozilla-mobile/android-components/blob/cafbf982760cd5a81b49d59a9f118f26a8bc87a7/components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt#L80-L120

Reply:
Awesome!